recipes/hole2/build.sh

```diff
@@ -0,0 +1,6 @@
+# compile HOLE with gfortran
+cd ${SRC_DIR}/src
+source ../source.apache
+
+make FC=${GFORTRAN} CC=${CC_FOR_BUILD}
+make FC=${GFORTRAN} CC=${CC_FOR_BUILD} PREFIX=${PREFIX} install-all
```

Comment:
I think we need to set more variables here so we are using the ones set by the conda-forge compilers infrastructure.

[All of these](https://github.com/osmart/hole2/blob/b8fdc748df2da19867a0b21778fd1115a118dbb2/src/Makefile#L20-L32) should (at least) include the ones the [activation scripts are setting at build time](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=653001&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=846):

```shell
# excerpt
...
+FFLAGS=-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/hole2-2.3.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix
...
```

Reply:
If I'm understanding correctly, for lines 5 and 6, all we need to do is add `FFLAGS=${FFLAGS}` and `CFLAGS=${CFLAGS}`. Is that what you're referring to?

Reply:
Can you just use appending `+=` instead of assignment `:=`
```make
FFLAGS += -fd-lines-as-comments -fbackslash -std=legacy -static-libgfortran
CFLAGS += -static-libgfortran
```
to ensure that our options are added to whatever the environment sets (given that [make turns env vars into makevars](https://www.gnu.org/software/make/manual/html_node/Environment.html)?

Reply:
My other question is what to do with program names such as `ar`. The current makefile has https://github.com/osmart/hole2/blob/b8fdc748df2da19867a0b21778fd1115a118dbb2/src/Makefile#L28 
```
AR := ar -rcv
```
and the [build env sets](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=653001&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=852)
```
AR=$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-ar
```
I think we need to **rewrite the HOLE Makefile** to *not include the `-rcv` in the command name* and instead put it after invocation like https://github.com/osmart/hole2/blob/b8fdc748df2da19867a0b21778fd1115a118dbb2/src/Makefile#L163-L169
```
$(LIB_NAME): $(FILES) 
	$(AR) -rcv $(LIB_NAME) $(FILES)  
	$(RANLIB) $(LIB_NAME)

.f.o:
	$(FC) $(FFLAGS) -c $*.f
	$(AR) -rcv $(LIB_NAME) $*.o
```
I am not 100% sure if we then just pass the build system's env vars explicitly to the Makefile as
```bash
make ... AR=${AR} ...
```
or if make will automatically take it from the env and use it instead of `AR := ar`.

Reply:
Final comment: in the Makefile we have a few invocations of `$(FC)` such as https://github.com/osmart/hole2/blob/b8fdc748df2da19867a0b21778fd1115a118dbb2/src/Makefile#L195-L196
```makefile
makver.exe: makver.f
	$(FC) makver.f -o makver.exe
```
which currently ignore FFLAGS and friends. Given that the build system insists on setting them, we also need to respect them so to be safe, the above should become
```makefile
makver.exe: makver.f
	$(FC) $(FFLAGS) $(LFLAGS) makver.f -o makver.exe
```
(Need to be checked if it works with the full set of FFLAGS...)